### DISCUSSION TITLE
Redesigning Backend for Poker Money Counting System

### DISCUSSION THREAD

**Discussion by kao-offline** [community/community#148099] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi everyone,

I recently decided to create a poker money counting system, but backend development isn’t really my strong suit. To help with this, I worked with an AI to build the backend. The current version works and includes all the required functionality, but I realized during testing that the game lobbies are being managed using local storage.

This is a problem because I want to make the system scalable and ideally host it on Vercel, which requires the backend to handle lobbies properly, without relying on local storage.

I’m looking for someone who might be willing to help me refactor or redesign the backend to fix this issue. The project is functional as-is, so you’ll have a clear starting point to work from.

Here’s the link to the current project version: https://github.com/KikoStudios/project/tree/dbbf7a680f6478e436fb9c49a7b8c73216195610

I’d greatly appreciate any help or guidance! Thanks in advance.

PS: I have no idea what i a doing please help

**Comment by divyamlathiya** [comment]
## Here's a suggested approach:

**Steps to Refactor the Backend:**

1. **Switch to a Persistent Database:**
Instead of relying on local storage, use a cloud database such as Firebase, MongoDB Atlas, or Google Cloud Firestore to store game lobby data. This will allow for easy scaling and persistence across multiple users and sessions.

2. **Replace Local Storage with Server-Side State:**
You'll need to store game state (lobbies, player info, etc.) on the server side. This can be done using a RESTful API or WebSocket connections, depending on your needs.

- **For REST API:** Use a framework like Express.js with a cloud database to manage lobby states (e.g., create, join, and leave lobbies).
- **For real-time updates:** You can use WebSockets or Socket.io for real-time communication and lobby updates between clients.

3. **Implement Lobby Management:**
Create API endpoints to handle lobby creation, joining, and leaving. For instance:

- `POST /lobbies` to create a new lobby.
- `GET /lobbies/{id}` to retrieve lobby data.
- `POST /lobbies/{id}/join` to allow a player to join a lobby.
- `DELETE /lobbies/{id}` to remove a lobby when it’s no longer needed.

4. **Deploy on Vercel:**
Vercel is great for hosting frontends but doesn’t provide traditional backend servers. You can use Serverless Functions to handle the backend logic and integrate them with your cloud database. Serverless functions (e.g., **Next.js API Routes** or **AWS Lambda**) can provide the necessary backend endpoints.

**Example Refactor:**
- **Backend:** Replace local storage management with a database (e.g., MongoDB):

```
// Example using Express and MongoDB for lobby management
const express = require('express');
const mongoose = require('mongoose');
const app = express();

// Lobby schema
const lobbySchema = new mongoose.Schema({
  name: String,
  players: [String],
});

const Lobby = mongoose.model('Lobby', lobbySchema);

app.post('/lobbies', async (req, res) => {
  const newLobby = new Lobby({ name: req.body.name, players: [] });
  await newLobby.save();
  res.status(201).send(newLobby);
});

app.post('/lobbies/:id/join', async (req, res) => {
  const lobby = await Lobby.findById(req.params.id);
  lobby.players.push(req.body.playerName);
  await lobby.save();
  res.send(lobby);
});

// Connect to MongoDB and start the app
mongoose.connect('mongodb://your-db-url')
  .then(() => app.listen(3000))
  .catch(err => console.error(err));
```

- **Frontend:** Use WebSockets or HTTP requests to interact with the backend.

**Resources:**
- [MongoDB Atlas](https://www.mongodb.com/cloud/atlas)
- [Socket.io](https://socket.io/)
- [Express.js](https://expressjs.com/)

By making these changes, you'll have a scalable backend that can handle game lobbies more effectively and can be easily deployed on platforms like Vercel. Let me know if you need further clarification!
